File: 0242_valid_anagram/is_anagram.c
#include <stdio.h>

/**
 * Time: O(n) 
 * Space: O(1)
 */
int is_anagram(char *s, char *t)
{
    if (strlen(s) != strlen(t))
        return 0;
    int alpha[26] = {};
    for (int i = 0; i < strlen(s); i++)
    {
        alpha[s[i] - 'a']++;
        alpha[t[i] - 'a']--;
    }
    for (int i = 0; i < 26; i++)
        if (alpha[i] > 0)
            return 0;
    return 1;
}

File: 0242_valid_anagram/is_anagram.go
package main

/**
 * Time: O(n)
 * Space: O(1)
 */
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	alphabet := make([]int, 26)
	for i := 0; i < len(s); i++ {
		alphabet[s[i]-'a']++
		alphabet[t[i]-'a']--
	}
	for i := 0; i < 26; i++ {
		if alphabet[i] > 0 {
			return false
		}
	}
	return true
}

func main() {

}
